src/tv2-common/content/dve.ts

```diff
@@ -204,68 +207,23 @@ export function MakeContentDVE2<
 
 	const classes: string[] = []
 
-	inputs.forEach(source => {
-		const sourceProps = source.split(':')
-		const fromCue = sourceProps[1]
-		const targetBox = Number(sourceProps[0])
-		if (!fromCue || !targetBox || isNaN(targetBox)) {
-			context.notifyUserWarning(`Invalid DVE mapping: ${sourceProps}`)
-			return
-		}
+	makeBoxMap(inputs, context, classes, dveGeneratorOptions, sources, boxMap)
 
-		classes.push(`${fromCue.replace(/\s/g, '')}_${dveGeneratorOptions.boxMappings[targetBox - 1]}`)
-
-		if (sources) {
-			const prop = sources[fromCue as keyof DVESources]
-			if (prop) {
-				boxMap[targetBox - 1] = prop.sourceType !== SourceType.SERVER || videoId ? prop : undefined
-			} else {
-				context.notifyUserWarning(`Missing mapping for ${targetBox}`)
-				boxMap[targetBox - 1] = undefined
-			}
-		} else {
-			// Need something to keep the layout etc
-			boxMap[targetBox - 1] = undefined
-		}
-	})
-
-	const boxes = _.map(template.boxes, box => ({ ...box, source: config.studio.AtemSource.Default }))
+	const boxes: BoxConfig[] = _.map(template.boxes, box => ({ ...box, source: config.studio.AtemSource.Default }))
```

Comment:
What is the reason that we are using the Map function from 'Underscore' instead of just using the Map function that Typescript provides directly on an Array?

In my opinion I think we should move away from using the 'Underscore' library.

Reply:
Aight, so I can see that `templates.boxes` isn't an array but:
```
boxes: {
		[key: number]: DVEConfigBox
	}
```
which I guess why we use the 'Underscore' library.

Then my question is, why don't we use `Map<number, DVEConfigBox>` as a type then?

I have seen this syntax quite a few places.

Reply:
It's parsed from JSON, so you would need to convert it to the Map. Is that needed? I don't think so if it's being used in one place only